### PR TITLE
A wizard for adding authentication devices

### PR DESCRIPTION
**Pull request by bl-nero**
Contributes to #36232 
Contributes to #37616 

This is a first pass at implementing the new designs. This PR covers all of the logic and interactions needed for the passkey dialog. Follow-up PRs will address layout and styling, as well as adding MFAs.

[Figma designs](https://www.figma.com/file/8DQZl1mYUPLgvKgmA6zPOM/In-Progress-Work?type=design&node-id=7052-56894&mode=design&t=DxwciTM9fGYvG46J-0)

[Demo](https://www.loom.com/share/69339a774cbf4a829b5c88893cc988d8?sid=0fe0133f-ee64-4c0e-98c8-ae1e83b14935)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -299,7 +299,7 @@ $(BUILDDIR)/teleport: ensure-webassets bpf-bytecode rdpclient
 TELEPORT_ARGS ?= start
 .PHONY: teleport-hot-reload
 teleport-hot-reload:
-	CompileDaemon --graceful-kill=true --exclude-dir=".git" --exclude-dir="node_modules" --build="make $(BUILDDIR)/teleport" --command="$(BUILDDIR)/teleport $(TELEPORT_ARGS)"
+	CompileDaemon --graceful-kill=true --exclude-dir=".git" --exclude-dir="node_modules" --exclude-dir="target" --build="make $(BUILDDIR)/teleport" --command="$(BUILDDIR)/teleport $(TELEPORT_ARGS)"
```
Comment (zmb3) [inline Makefile:302]: This seems completely unrelated. I'd move it into a separate PR.

**Review by zmb3 — COMMENTED** [review]
Code and tests look good, but I'm not sure we should merge this until the styling is fixed. It's a relatively small PR as-is, and by merging what you have here we introduce a chance that we fail to backport styling updates and show users a set of screens that look half done.

**Comment by bl-nero** [comment]
> Code and tests look good, but I'm not sure we should merge this until the styling is fixed. It's a relatively small PR as-is, and by merging what you have here we introduce a chance that we fail to backport styling updates and show users a set of screens that look half done.

@zmb3 Good catch, I forgot to put this change behind the flag that I prepared last week. Fixed it (it was just one statement that triggers the flow; the rest of the logic just lives there in parallel). The thing is, I'd still like to push this change through, since it's sizable enough, and I'd like to first integrate both passkeys and MFAs to the new flow, and only then work on the styling details.

File: web/packages/teleport/src/Account/ManageDevices/AddAuthDeviceWizard/AddAuthDeviceWizard.test.tsx [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+  jest
+    .spyOn(auth, 'createPrivilegeTokenWithTotp')
+    .mockImplementationOnce(token =>
+      Promise.resolve(`totp-privilege-token-${token}`)
+    );
+});
+
+afterEach(jest.resetAllMocks);
+
+interface TestWizardProps {
```
Comment (gzdunek) [inline web/packages/teleport/src/Account/ManageDevices/AddAuthDeviceWizard/AddAuthDeviceWizard.test.tsx:54]: It is used in a single place, I'd line it.

File: web/packages/teleport/src/Account/ManageDevices/AddAuthDeviceWizard/AddAuthDeviceWizard.tsx [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+                clearAttempt();
+              }}
+            />
+            {mfaOption === 'otp' && (
+              <FieldInput
+                label="Authenticator Code"
+                rule={requiredField('Authenticator code is required')}
+                inputMode="numeric"
+                autoComplete="one-time-code"
+                value={authCode}
```
Comment (gzdunek) [inline web/packages/teleport/src/Account/ManageDevices/AddAuthDeviceWizard/AddAuthDeviceWizard.tsx:184]: I'm wondering if we should introduce a new name, I believe we use 'otp code' everywhere else.

**Review by gzdunek — APPROVED** [review]
Looks good, I left some code-style comments.

File: web/packages/teleport/src/Account/ManageDevices/AddAuthDeviceWizard/AddAuthDeviceWizard.tsx (hunk 3 above)
Comment (bl-nero) [reply web/packages/teleport/src/Account/ManageDevices/AddAuthDeviceWizard/AddAuthDeviceWizard.tsx:184]: No, there's actually a discrepancy between the Web UI and the CLI. The change is consistent with what were doing until now.

**Review by ryanclark — APPROVED** [review]

**Comment by bl-nero** [comment]
@gzdunek FYI, I decided to leave the error message in the backend as it is, since it's consistent with CLI, and instead translate it in the UI code (and clearly mark dependencies on both sides so it's visible where we depend on it).
